controllers/storagecluster/cephcluster.go

```diff
@@ -1341,3 +1345,31 @@ func getOsdCount(sc *ocsv1.StorageCluster) int {
 	}
 	return osdCount
 }
+
+func determineDefaultCephDeviceClass(foundDeviceClasses []rookCephv1.DeviceClasses, isReplica1 bool, replica1DeviceClasses []string) string {
+	// If non-resilient pools(replica-1) is not enabled
+	if !isReplica1 {
+		// If device classes are found, return the first one
+		if len(foundDeviceClasses) != 0 {
+			return foundDeviceClasses[0].Name
+		}
+		// If device classes are not found, return "ssd"
+		return "ssd"
+	}
+
+	// If non-resilient pools(replica-1) is enabled
+	// If device classes are found, return the first one that is not a replica1 device class
+	if len(foundDeviceClasses) != 0 {
+		replica1DeviceClassMap := make(map[string]bool)
+		for _, deviceClass := range replica1DeviceClasses {
+			replica1DeviceClassMap[deviceClass] = true
+		}
+		for _, deviceClass := range foundDeviceClasses {
+			if found := replica1DeviceClassMap[deviceClass.Name]; !found {
+				return deviceClass.Name
+			}
+		}
+	}
+	// If device classes are not found, return "replicated"
+	return "replicated"
```

Comment:
How about if we always default to `ssd`? We will only use `replicated` for existing clusters that find it above.
```suggestion
	return "ssd"
```

Reply:
We can change it to ssd, but then we also need to remove this https://github.com/red-hat-storage/ocs-operator/blob/b18b234e14f6182c858e64bffd3e627b1a5350b3/controllers/storagecluster/cephcluster.go#L850-L853

Reply:
Ok good catch, seems like we anyway want to set that annotation to the DefaultDeviceClass.

Reply:
Modified